### PR TITLE
Splitting backend and frontend into separate yarn packages

### PR DESCRIPTION
**Pull request by yradsmikham**
Closes https://github.com/microsoft/bedrock/issues/1353.

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "build": "tsc src/** --outDir ../build/backend/ && tsc src/server.ts --outDir ../build/",
+    "build-prod": "tsc src/** --outDir ./build/",
+    "lint": "tslint -p tsconfig.json -c tslint.json src/*.ts"
+  },
+  "devDependencies": {
+    "@types/node": "^12.6.8",
+    "ts-node": "^8.5.4",
+    "tslint": "^5.20.0",
+    "tslint-config-prettier": "^1.18.0",
+    "tslint-react": "^4.2.0",
```
Comment (dennisseah) [inline backend/package.json:21]: do we need tslint-react for backend?

File: backend/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  "license": "MIT",
+  "dependencies": {
+    "@types/express": "^4.17.0",
+    "spektate": "^1.0.10",
+    "express": "^4.17.1"
+  },
+  "scripts": {
+    "start": "ts-node src/index.ts",
+    "build": "tsc src/** --outDir ../build/backend/ && tsc src/server.ts --outDir ../build/",
+    "build-prod": "tsc src/** --outDir ./build/",
```
Comment (dennisseah) [inline backend/package.json:13]: can you explain why we need to build  and build-prod? thanks

File: backend/package.json (hunk 1 above)
Comment (samiyaakhtar) [reply backend/package.json:21]: will remove, it was added as part of `depcheck` i believe.

File: backend/package.json (hunk 2 above)
Comment (samiyaakhtar) [reply backend/package.json:13]: `yarn build` is for local builds, and `yarn build-prod` is being used by dockerfile. I don't see a whole lot of use for `yarn build` for local builds though, so i could just remove `yarn build` and rename build-prod to build. `yarn build` is currently being run in pipelines but they can use either one.

**Pull request merged**